### PR TITLE
Connection serial tests

### PR DESCRIPTION
**Pull request by sacOO7**
- Fixed failing tests for no-connection-serial PR

### CODE REVIEW

**Comment by AndyTWF** [comment]
Can we get a description of the key changes in the PR, please :)

**Comment by sacOO7** [comment]
> Can we get a description of the key changes / issues fixed in the PR, please :)

updated!

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java [hunk 1]
```diff
@@ -1070,24 +993,21 @@ public void onConnectionStateChanged(ConnectionStateChange state) {
...
+            assertEquals("Second round of messages (queued) has errors", 0, resendErrors.length);
 
-            for (PresenceMessage presenceMessage:
-                transport.getSentPresenceMessages()) {
+            for (PresenceMessage presenceMessage: transport.getSentPresenceMessages()) {
                 System.out.println("presence_resume_test: sent message with client: "+presenceMessage.clientId +" " +
                     " action:"+presenceMessage.action);
             }
-            assertEquals("Second round of messages has incorrect size", 9, transport.getSentPresenceMessages().size());
+            assertEquals("Second round of messages has incorrect size", 6, transport.getSentPresenceMessages().size());
```
Comment (sacOO7) [inline lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java:1002]: @AndyTWF can you please go through this test once. This test is passing with expected value as `9` on `main` branch. I feel, the right value should be `6`. So, as of now it's passing with value of `6` on `no-connection-serial` branch.

Comment (sacOO7) [reply lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java:1002]: Seems there might be a bug before that got fixed while working on `no-connection-serial` spec

Comment (ttypic) [reply lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java:993]: Agree, it looks like it should be 6

**Review by ttypic — APPROVED** [review]
Looks good to me

File: lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java (hunk 1 above)
Comment (sacOO7) [reply lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java:993]: Thanks !

File: lib/src/test/java/io/ably/lib/test/common/Helpers.java [hunk 2]
```diff
@@ -401,6 +408,48 @@ public PresenceMessage contains(String clientId, String connectionId, PresenceMe
...
+            this.connectionManager = ablyRealtime.connection.connectionManager;
+        }
+
+        public void setField(String fieldName, long value) {
+            try {
+                Field connectionStateField = ConnectionManager.class.getDeclaredField(fieldName);
+                connectionStateField.setAccessible(true);
+                connectionStateField.setLong(connectionManager, value);
+            } catch (NoSuchFieldException | IllegalAccessException e) {
+                fail("Failed updating " + fieldName + " with error " + e);
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply lib/src/test/java/io/ably/lib/test/common/Helpers.java:424]: Though I agree, failing something doesn't fall under helper classes. Feels out of context

**Pull request merged**